cheatsheets/Cryptographic_Storage_Cheat_Sheet.md

```diff
@@ -1,234 +1,185 @@
+- Simplifying compliance with regulatory standards such as FIPS 140 or PCI DSS.
+- Making it harder for an attacker to export or steal keys.
 
-###### 3.6.6 Split knowledge and establishment of dual control of cryptographic keys
+In some cases none of these will be available, such as in a shared hosting environment, meaning that it is not possible to obtain a high degree of protection for any encryption keys. However, the following basic rules can still be followed:
 
-The requirement for split knowledge and/or dual control for key management prevents an individual user performing key management tasks such as key rotation or deletion. The system should require two individual users to perform an action (i.e. entering a value from their own OTP) which creates to separate values which are concatenated to create the final key data.
+- Do not hard-code keys into the application code.
+- Do not check keys into version control systems.
+- Protect the configuration files containing the keys with restrictive permissions.
+- Avoid storing keys in environment variables, as these can be accidentally exposed through functions such as [phpinfo()](https://www.php.net/manual/en/function.phpinfo.php) or through the `/proc/self/environ` file.
 
-###### 3.6.7 Prevention of unauthorized substitution of cryptographic keys
+### Separation of Keys and Data
 
-The system put in place to comply with requirement 3.6.6 can go a long way to preventing unauthorised substitution of key data. In addition to the dual control process you should implement strong access control, auditing and logging for key data so that unauthorised access attempts are prevented and logged.
+Where possible, encryption keys should be stored in a separate location from encrypted data. For example, if the data is stored in a database, the keys should be stored in the filesystem. This means that if an attacker only has access to one of these (for example through directory traversal or SQL injection), they cannot access both the keys and the data.
 
-###### 3.6.8 Requirement for cryptographic key custodians to sign a form stating that they understand and accept their key-custodian responsibilities
+Depending on the architecture of the environment, it may be possible to store the keys and data on separate systems, which would provide a greater degree of isolation.
 
-To perform the strong key management functions we have seen in requirement 3.6 we must have highly trusted and trained key custodians who understand how to perform key management duties. The key custodians must also sign a form stating they understand the responsibilities that come with this role.
+### Encrypting Stored Keys
 
-# Related documentation & tools
+Where possible, encryption keys should themselves be stored in an encrypted form. At least two separate keys are required for this:
 
-## Documentation
+- The Data Encryption Key (DEK) is used to encrypt the data.
+- The Key Encryption Key (KEK) is used to encryption the DEK.
```

Comment:
Change 'encryption' to 'encrypt'. Small grammatical change.

Reply:
Fixed